microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact.md

```diff
@@ -0,0 +1,251 @@
+
+Use the following best practices when configuring where your content scripts are loaded:
+
+* Use the most specific URL patterns possible for the `matches` and `exclude_matches` properties in your extension manifest file. For example, if your content script only needs to run on webpages of the example.com domain, use `https://example.com/*` instead of "`*://*/*`.
+* To control whether your content script only runs in the top-level frame, or also in nested frames of the webpage that matches a URL pattern, use the `all_frames` property in your extension manifest file. By default, this property is set to false, which means that your content script will only run in the top-level frame. If your content script needs to access or modify the DOM in nested frames, set this property to true. Note that setting `all_frames` to true increases the amount of code that runs on a webpage.
+
+
+<!-- ====================================================================== -->
+## Load content scripts only when needed
+
+To reduce the amount of code that is loaded and that runs on each webpage, and to save memory and CPU resources, only load your content scripts as needed, instead of on every page load.
+
+
+<!-- ------------------------------ -->
+#### Configure when to load content scripts in your extension manifest file
+
+To control when your extension's content script should be loaded, use the `run_at` property in your extension's manifest file.
+
+By default, this property is set to the `document_idle` value, which means that the content script will be loaded, and run, after the page has finished loading and the DOM is ready. This is the recommended value for most content scripts. The `document_idle` value ensures that the content script will not interfere with the page loading process.
+
+To load and run your content script before the page is fully loaded, use the `document_start` or `document_end` values. These values are useful in cases such as modifying the webpage layout or styling, but they can also cause performance issues or compatibility problems with other scripts on the page.
+
+
+<!-- ------------------------------ -->
+#### Programmatically load content scripts at runtime
+
+To programmatically load your content scripts at runtime, only when needed, use the `chrome.scripting` API . The `chrome.scripting` API provides more control over when and where your content script is loaded.
+
+For example, you can use the `chrome.scripting` API to load your content script only after the user has interacted with the webpage or extension UI, such as when clicking on an extension button, or a part of the webpage.
```

Comment:
```suggestion
For example, you can use the `chrome.scripting` API to load your content script only after the user has interacted with the webpage or extension UI, such as when clicking an extension button, or a part of the webpage.
```
according to style guide, not "click on" ... actually style guide lacks any direct guidance, and says to write in an abstract way (🧂) that applies to mouse and keyboard and doesn't differentiate them (vs. other co. style guides).  
style guide does have comparable example "right-click the foo".